src/main/java/ch/njol/skript/conditions/CondCompare.java

```diff
@@ -251,22 +252,53 @@ private boolean init(String expr) {
 	 * @return A literal value, or null if parsing failed.
 	 */
 	@Nullable
-	private <T> SimpleLiteral<T> reparseLiteral(Class<T> type, Expression<?> expr) {
-		if (expr instanceof SimpleLiteral) { // Only works for simple literals
-			Expression<?> source = expr.getSource();
-			
-			// Try to get access to unparsed content of it
-			if (source instanceof UnparsedLiteral) {
-				String unparsed = ((UnparsedLiteral) source).getData();
-				T data = Classes.parse(unparsed, type, ParseContext.DEFAULT);
-				if (data != null) { // Success, let's make a literal of it
-					return new SimpleLiteral<>(data, false, new UnparsedLiteral(unparsed));
-				}
+	private <T> SimpleLiteral<T> reparseLiteral(Class<T> type, Expression<?> expression) {
+		Expression<?> source = expression;
+		if (expression instanceof SimpleLiteral) // Only works for simple literals
+			source = expression.getSource();
+
+		// Try to get access to unparsed content of it
+		if (source instanceof UnparsedLiteral) {
+			String unparsed = ((UnparsedLiteral) source).getData();
+			T data = Classes.parse(unparsed, type, ParseContext.DEFAULT);
+			if (data != null) { // Success, let's make a literal of it
+				return new SimpleLiteral<>(data, false, new UnparsedLiteral(unparsed));
 			}
 		}
 		return null; // Context-sensitive parsing failed; can't really help it
 	}
-	
+
+	/**
+	 * Attempts to transform an UnparsedLiteral into a type that is comparable to another.
+	 * For example 'fire' will be VisualEffect without this, but if the user attempts to compare 'fire'
+	 * with a block. This method will see if 'fire' can be compared to the block, and it will find ItemType.
+	 * Essentially solving something a human sees as comparable, but Skript doesn't understand.
+	 * 
+	 * @param one The UnparsedLiteral expression to attempt to reconstruct.
+	 * @param two any expression to grab the return type from.
+	 * @return The newly formed Literal, will be SimpleLiteral in most cases.
+	 */
+	@SuppressWarnings("unchecked")
+	private Literal<?> attemptReconstruct(UnparsedLiteral one, Expression<?> two) {
+		Expression<?> expression = null;
+		// Must handle numbers first.
+		expression = one.getConvertedExpression(Number.class);
+		if (expression == null) {
+			for (ConverterInfo<?, ?> converter : Converters.getConverterInfos()) {
+				// We're comparing the 'two' expression, so we need the 'from' for comparing.
+				if (!converter.getFrom().isAssignableFrom(two.getReturnType()))
+					continue;
+				// Must be comparable if we're going to attempt to convert this UnparsedLiteral. Otherwise why attempt something that won't work.
+				if (Comparators.getComparator(two.getReturnType(), converter.getTo()) == null)
+					continue;
+				expression = reparseLiteral(converter.getTo(), one);
+			}
+		}
```

Comment:
Maybe I'm misunderstanding this code, but it seems like you could just call `Comparators#getComparatorInfo` here and use the `getTo` of the result (if one is found)

Reply:
No, the code needs to check for a valid solution of the `from` and the `to` to attempt against it. It already does that check earlier.

Reply:
Sorry, I had a bit of a misunderstanding of the code in my initial comment lol.

I think we could loop over all of the classinfos with parsers and see if their types have a comparator with `two`. The `getComparator` method should check if Converters can be chained to make it to the desired type.